pkg/controller/clone-controller.go

```diff
@@ -315,30 +315,30 @@ func (c *CloneController) processPvcItem(pvc *v1.PersistentVolumeClaim) error {
 	}
 
 	if needsSync && (sourcePod == nil || targetPod == nil) {
+		createPodCount := 0
+
 		//create random string to be used for pod labeling and hostpath name
 		if sourcePod == nil {
 			cr, err := getCloneRequestPVC(pvc)
 			if err != nil {
 				return err
 			}
-			// all checks passed, let's create the cloner pods!
-			c.expectPodCreate(pvcKey)
 			//create the source pod
 			sourcePod, err = CreateCloneSourcePod(c.clientset, c.cloneImage, c.verbose, c.pullPolicy, cr, pvc)
 			if err != nil {
-				c.observePodCreate(pvcKey)
 				return err
 			}
+			createPodCount++
 		}
 		if targetPod == nil {
-			c.expectPodCreate(pvcKey)
 			//create the target pod
 			targetPod, err = CreateCloneTargetPod(c.clientset, c.cloneImage, c.verbose, c.pullPolicy, pvc, sourcePod.ObjectMeta.Namespace)
 			if err != nil {
-				c.observePodCreate(pvcKey)
 				return err
```

Comment:
one thing to point out. This return will cause issues if the source pod creation passes and the target pod creation fails.  We won't have an expectation set for the src pod.